src/dbus/server/dbus_object.hpp

```diff
@@ -47,17 +47,30 @@
 
 #include <dbus/dbus.h>
 
+#include "border_agent/border_agent.hpp"
 #include "common/code_utils.hpp"
 #include "common/types.hpp"
 #include "dbus/common/constants.hpp"
 #include "dbus/common/dbus_message_dump.hpp"
 #include "dbus/common/dbus_message_helper.hpp"
 #include "dbus/common/dbus_resources.hpp"
 #include "dbus/server/dbus_request.hpp"
+#include "host/thread_host.hpp"
+#include "mdns/mdns.hpp"
 
 namespace otbr {
 namespace DBus {
 
+class DependentComponents
+{
+public:
+    Host::ThreadHost &mHost;
+    Mdns::Publisher  &mPublisher;
+#if OTBR_ENABLE_BORDER_AGENT
+    otbr::BorderAgent &mBorderAgent;
+#endif
```

Comment:
This could make the implementation diverging. Could we always enable border agent?

@bukepo

Reply:
What does diverging mean? It makes sense to me to always enable border agent. But then I think we'd better remove the flag 'OTBR_ENABLE_BORDER_AGENT'. Thoughts?

Reply:
By diverge I meant there'll always be two variants of code (enable/disable BA) being used.

> But then I think we'd better remove the flag 'OTBR_ENABLE_BORDER_AGENT'.

Yes. That's also what I think we should do eventually. In this way there's no risk of diverging.

Reply:
That sounds good to me. Maybe let's do it in a seperate PR?

Reply:
> That sounds good to me. Maybe let's do it in a seperate PR?

SGTM